### ISSUE TITLE
Opening types menu causes hang, massive memory allocation

### ISSUE THREAD

**Issue by Douggem** [Vector35/binaryninja-api#3761]
**Version and Platform (required):**
- Binary Ninja Version: 3.2.3945-dev] 
- OS: Windows
- CPU Architecture AMD64

**Bug Description:**
Attempting to open the types window causes Binja to hang and continuously allocate memory

This is after auto-generating ~15,000 structures from RTTI data.  Everything else seems to run smooth, including using the generated structures in disassembly, until opening the types menu.

The offending bndb, with the auto-generated structures in it, can be downloaded from https://drive.google.com/file/d/1L-L7-lV-jIaUpRc6acKsUWL5tKaruTh0/view?usp=sharing

FWIW, in a previous step I auto-generated the same structures but without any members and the types menu opened OK.

**Steps To Reproduce:**
1. Open this BNDB
2. Open the types menu
3. Pain

**Comment by plafosse** [comment]
This is a bug caused by the `UADef` structure somehow having a member with an offset of `0xfffffffffffffff4` causing us to try to render all these. The weird thing is that the structure is reporting a normal size. I haven't fully diagnosed the issue but if you undefine that structure in the console you'll probably have a much better time. Also perhaps you RTTI recovery stuff has a bug in it some how.  If you find the bug in your stuff then please let us know what it was so we can fix that API. Still root causing this for now though ....

**Comment by plafosse** [comment]
Here's a little script to find and mitigate issues like this:
```
>>> for n,t in bv.types:
...  if isinstance(t, StructureType):
...   for m in t.members:
...    if m.offset > len(t):
...     bv.undefine_user_type(n)
...     break
```

**Comment by Douggem** [comment]
Thanks a lot Peter.  I'm recovering types through RTTI then using my own structure maker script to fill out the types based on usage.  I am not checking for negative offsets when filling out the struct.  So I'm hitting this bit of code:

![image](https://user-images.githubusercontent.com/4275220/210148508-1a0a6b64-7192-46b2-aee2-71109d576645.png)

Seeing the *rax_5 = &vft_UADef__0, and defining rax_5 as a UADef, not differentiating between the different usages of rax_5, and then assigning members at those negative offsets

![image](https://user-images.githubusercontent.com/4275220/210148546-ac9dd243-1cf5-4ba7-941d-bd8e7bcc4d93.png)

So definitely a bug on my end.  Thanks again!

**Comment by galenbwill** [comment]
Should be fixed in builds >= 3971 (See #3600)
